mypy/checker.py

```diff
@@ -2745,46 +2745,46 @@ def check_multiple_inheritance(self, typ: TypeInfo) -> None:
             # No multiple inheritance.
             return
         # Verify that inherited attributes are compatible.
-        mro = typ.mro[1:]
-        for i, base in enumerate(mro):
+        bases = typ.bases
+        all_names = [{n for p in b.type.mro for n in p.names} for b in bases]
```

Comment:
I don't think, that is enough. Mixing MRO and Bases looks confusing, also I am not sure, how it works for nested generics inheritance

Could you please take a look at my suggestion in #18274 ? I think, we can find the better solution together